File: Queue/Q2. Time Needed to Buy Tickets/Q2_TimeNeededtoBuyTickets.py
class Solution(object):                          # Define the Solution class
    def timeRequiredToBuy(self, tickets, k):     # Method to calculate total time
        """
        :type tickets: List[int]                 # List of tickets needed by each person
        :type k: int                             # Index of the target person
        :rtype: int                              # Return total time
        """
        c = 0                                    # Counter to store time spent
        
        while True:                              # Run until target person finishes
            if tickets[k] == 0:                  # If target person has no tickets left
                break                            # Stop the loop
            
            for i in range(len(tickets)):        # Loop through all people
                if tickets[k] == 0:              # If target person is done
                    break                        # Stop the loop
                if tickets[i] > 0:               # If current person needs tickets
                    tickets[i] -= 1              # Give one ticket
                    c += 1                       # Increase time by 1 second
        
        return c                                 # Return total time taken

File: Sequence Valley/Queue/Q2. Time Needed to Buy Tickets/Q2_TimeNeededtoBuyTickets.java
class Solution {                                      // Define the Solution class
    public int timeRequiredToBuy(int[] tickets, int k) { // Method to calculate total time
        
        int total = 0;                                // Store total time taken

        for (int i = 0; i < tickets.length; i++) {    // Loop through each person
            if (i <= k) {                             // If person is before or at k
                total += Math.min(tickets[i], tickets[k]); // Add minimum possible seconds
            } else {                                  // If person is after k
                total += Math.min(tickets[i], tickets[k] - 1); // Add reduced time
            }
        }

        return total;                                 // Return total time needed
    }
}
